src/Transport/Topology/MetaModel/EntityInfo.cs

```diff
@@ -23,17 +22,12 @@ public EntityInfo()
 
         protected bool Equals(EntityInfo other)
         {
-            if (Type == EntityType.Subscription)
-            {
-                var entity = RelationShips.First(r => r.Type == EntityRelationShipType.Subscription);
-                var otherEntity = other.RelationShips.First(r => r.Type == EntityRelationShipType.Subscription);
-                var targetPathEquals = string.Equals(entity.Target.Path, otherEntity.Target.Path);
-                if (!targetPathEquals)
-                {
-                    return false;
-                }
-            }
-            return string.Equals(Path, other.Path) && Type == other.Type && Equals(Namespace, other.Namespace);
+            return string.Equals(Path, other.Path) && Type == other.Type && Equals(Namespace, other.Namespace) && DerivedEqual(other);
```

Comment:
I'm ok with this change as it does not break the backward compatibility. 

_Reasoning:_
If somebody used the `SubscriptionInfo.Equals` before, it wasn't working properly. This changes the behavior making is proper. The change is not breaking until somebody emitted a call directly against `EntityInfo.Equals` which is highly unlikely. Having this said, we're not breaking anything.